src/extension/android/androidPlatform.ts

```diff
@@ -187,21 +185,22 @@ export class AndroidPlatform extends GeneralMobilePlatform {
         });
     }
 
-    private launchAppWithADBReverseAndLogCat(device: IDevice): Q.Promise<void> {
-        return Q({})
+    private launchAppWithADBReverseAndLogCat(device: IDevice): Promise<void> {
+        return new Promise(() => {
+            return this.configureADBReverseWhenApplicable(device);
+            })
             .then(() => {
-                return this.configureADBReverseWhenApplicable(device);
-            }).then(() => {
                 return this.needsToLaunchApps
                     ? this.adbHelper.launchApp(this.runOptions.projectRoot, this.packageName, device.id)
-                    : Q<void>(void 0);
-            }).then(() => {
+                    : Promise.resolve();
```

Comment:
Could it be better to use the `resolve ()` function instead of `Promise.resolve()`?

Reply:
We cannot pass `resolve` function from the top Promise to `then`.